openmc/data/reaction.py

```diff
@@ -0,0 +1,1189 @@
+
+        Returns
+        -------
+        rx : openmc.data.Reaction
+            Reaction data
+
+        """
+        rx = Reaction(mt)
+
+        # Integrated cross section
+        if (3, mt) in ev.section:
+            file_obj = StringIO(ev.section[3, mt])
+            get_head_record(file_obj)
+            params, rx.xs['0K'] = get_tab1_record(file_obj)
+            rx.q_value = params[1]
+
+        # Get fission product yields (nu) as well as delayed neutron energy
+        # distributions
+        if mt in (18, 19, 20, 21, 38):
+            rx.products, rx.derived_products = _get_fission_products_endf(ev)
+
+        if (6, mt) in ev.section:
+            # Product angle-energy distribution
+            for product in _get_products(ev, mt):
+                if mt in (18, 19, 20, 21, 38) and product.particle == 'neutron':
+                    rx.products[0].applicability = product.applicability
+                    rx.products[0].distribution = product.distribution
+                else:
+                    rx.products.append(product)
+
+        elif (4, mt) in ev.section or (5, mt) in ev.section:
+            # Uncorrelated angle-energy distribution
+            neutron = Product('neutron')
+
+            # Note that the energy distribution for MT=455 is read in
+            # _get_fission_products_endf rather than here
+            if (5, mt) in ev.section:
+                file_obj = StringIO(ev.section[5, mt])
+                items = get_head_record(file_obj)
+                nk = items[4]
+                for i in range(nk):
+                    params, applicability = get_tab1_record(file_obj)
+                    dist = UncorrelatedAngleEnergy()
+                    dist.energy = EnergyDistribution.from_endf(file_obj, params)
+
+                    neutron.applicability.append(applicability)
+                    neutron.distribution.append(dist)
+            elif mt == 2:
+                # Elastic scattering -- no energy distribution is given since it
+                # can be calulcated analytically
+                dist = UncorrelatedAngleEnergy()
+                neutron.distribution.append(dist)
+            elif mt >= 51 and mt < 91:
+                # Level inelastic scattering -- no energy distribution is given
+                # since it can be calculated analytically. Here we determine the
+                # necessary parameters to create a LevelInelastic object
+                dist = UncorrelatedAngleEnergy()
+
+                A = ev.target['mass']
+                threshold = (A + 1.)/A*abs(rx.q_value)
+                mass_ratio = (A/(A + 1.))**2
+                dist.energy = LevelInelastic(threshold, mass_ratio)
+
+                neutron.distribution.append(dist)
+
+            if (4, mt) in ev.section:
+                for dist in neutron.distribution:
+                    dist.angle = AngleDistribution.from_endf(ev, mt)
+
+            if mt in (18, 19, 20, 21, 38) and (5, mt) in ev.section:
```

Comment:
This goes for multiple instances in this PR... it might be a good idea to group things into global constants instead of checking against a tuple of ints.

For example:

```
if mt in (18, 19, 20, 21, 38)
```

vs

```
FISSION_REACTION_TYPES = (18, 19, 20, 21, 38)
if mt in FISSION_REACTION_TYPES
```

This allows for better readability as well as maintainability (in the case that you want to add a new type to this).